File: app/src/main/java/com/devidea/timeleft/DatabaseDao.java
package com.devidea.timeleft;

import androidx.room.Dao;
import androidx.room.Insert;
import androidx.room.Query;
import androidx.room.Update;

import java.util.ArrayList;
import java.util.List;

@Dao
public interface DatabaseDao {

    @Insert
    void saveItem(EntityItemInfo entityItemInfo);

    @Query("SELECT * FROM EntityItemInfo")
    List<EntityItemInfo> getItem();

    @Query("DELETE FROM EntityItemInfo WHERE id = :ID")
    void deleteItem(int ID);

    @Query("SELECT * FROM EntityItemInfo WHERE id = :ID")
    EntityItemInfo getSelectItem(int ID);

    @Query("UPDATE EntityItemInfo SET startValue = :updateStart, endValue = :updateEnd WHERE id = :ID")
    void updateItem(String updateStart, String updateEnd, int ID);

    //위젯엔티티

    @Insert
    void saveWidget(EntityWidgetInfo entityWidgetInfo);

    @Query("SELECT widgetID FROM EntityWidgetInfo")
    int[] get();

    @Query("SELECT type FROM EntityWidgetInfo WHERE widgetID = :ID")
    String getType(int ID);

    @Query("SELECT ItemID FROM EntityWidgetInfo WHERE widgetID = :ID")
    int getTypeID(int ID);

    @Query("DELETE FROM EntityWidgetInfo WHERE widgetID = :ID")
    void delete(int ID);

    @Query("DELETE FROM EntityWidgetInfo WHERE ItemID = :ID")
    void deleteCustomWidget(int ID);


}

File: app/src/main/java/com/devidea/timeleft/DatabaseDao.kt
package com.devidea.timeleft

import androidx.room.*

@Dao
interface DatabaseDao {
    @Insert
    fun saveItem(entityItemInfo: EntityItemInfo?)

    @get:Query("SELECT * FROM EntityItemInfo")
    val item: List<EntityItemInfo?>

    @Query("DELETE FROM EntityItemInfo WHERE id = :ID")
    fun deleteItem(ID: Int)

    @Query("SELECT * FROM EntityItemInfo WHERE id = :ID")
    fun getSelectItem(ID: Int): EntityItemInfo

    @Query("UPDATE EntityItemInfo SET startValue = :updateStart, endValue = :updateEnd WHERE id = :ID")
    fun updateItem(updateStart: String?, updateEnd: String?, ID: Int)

    //위젯엔티티
    @Insert
    fun saveWidget(entityWidgetInfo: EntityWidgetInfo?)

    @Query("SELECT widgetID FROM EntityWidgetInfo")
    fun get(): IntArray?

    @Query("SELECT type FROM EntityWidgetInfo WHERE widgetID = :ID")
    fun getType(ID: Int): String?

    @Query("SELECT ItemID FROM EntityWidgetInfo WHERE widgetID = :ID")
    fun getTypeID(ID: Int): Int

    @Query("DELETE FROM EntityWidgetInfo WHERE widgetID = :ID")
    fun delete(ID: Int)

    @Query("DELETE FROM EntityWidgetInfo WHERE ItemID = :ID")
    fun deleteCustomWidget(ID: Int)
}